### PR TITLE
Core/Spells: Fixed Wild Hunt

### PR DESCRIPTION
**Pull request by Goatform**
Closes issues #6709 and #12593

### CODE REVIEW

File: src/server/game/Entities/Pet/Pet.cpp [hunk 1]
```diff
@@ -948,6 +949,22 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
             //damage range is then petlevel / 2
             SetBaseWeaponDamage(BASE_ATTACK, MAXDAMAGE, float(petlevel + (petlevel / 4)));
             //damage is increased afterwards as strength and pet scaling modify attack power
+            if (Player* player = ToPlayer())
```
Comment (joschiwald) [inline src/server/game/Entities/Pet/Pet.cpp]: how can this work? this is the guardian and never a player

File: src/server/game/Entities/Pet/Pet.cpp [hunk 2]
```diff
@@ -948,6 +949,22 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
...
+            {
+                if (Pet* pet = player->GetPet())
+                {
+                    if (pet->GetAura(SPELL_WILD_HUNT_R2))
+                    {
+                        SetModifierValue(UNIT_MOD_STAT_STAMINA, BASE_VALUE, float(m_owner->GetStat(STAT_STAMINA)) * 0.4f);
+                        SetBonusDamage(int32(GetOwner()->GetTotalAttackPowerValue(BASE_ATTACK) * 0.3f));
```
Comment (joschiwald) [inline src/server/game/Entities/Pet/Pet.cpp]: use values from dbc

Comment (Shurenchuuun) [inline src/server/game/Entities/Pet/Pet.cpp]: Not BASE_ATTACK. 
RANGED_ATTACK.

Comment (tkrokli) [inline src/server/game/Entities/Pet/Pet.cpp]: Yes, line 959 and 964 needs the same info source as StatSystem.cpp line 1310 (R.A.P.).

File: src/server/game/Entities/Pet/Pet.cpp (hunk 1 above)
Comment (Goatform) [inline src/server/game/Entities/Pet/Pet.cpp]:
```
@Keader commented:
Fix By: @Goatform
#15712

This work for me

File: src/server/game/Entities/Pet/Pet.cpp (hunk 2 above)
Comment (Goatform) [inline src/server/game/Entities/Pet/Pet.cpp]: http://wotlk.openwow.com/spell=62758
http://wotlk.openwow.com/spell=62762

File: src/server/game/Entities/Pet/Pet.cpp (hunk 1 above)
Comment (ccrs) [inline src/server/game/Entities/Pet/Pet.cpp]: try it without this code, it makes no actual sense.

**Comment by Aokromes** [comment]
/home/travis/build/TrinityCore/TrinityCore/src/server/game/Entities/Pet/Pet.cpp:924:13: fatal error: 
      enumeration value 'MAX_PET_TYPE' not handled in switch [-Wswitch]
    switch (petType)
            ^

File: src/server/game/Entities/Pet/Pet.cpp [hunk 3]
```diff
@@ -948,8 +949,20 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
...
-            break;
+            if (GetAura(SPELL_WILD_HUNT_R2))
+            {
+                SetModifierValue(UNIT_MOD_STAT_STAMINA, BASE_VALUE, float(m_owner->GetStat(STAT_STAMINA)) * 0.4f);
+                SetBonusDamage(int32(GetOwner()->GetTotalAttackPowerValue(RANGED_ATTACK) * 0.3f));
+            }
+            else if (GetAura(SPELL_WILD_HUNT_R1))
+            {
+                SetModifierValue(UNIT_MOD_STAT_STAMINA, BASE_VALUE, float(m_owner->GetStat(STAT_STAMINA)) * 0.2f);
+                SetBonusDamage(int32(GetOwner()->GetTotalAttackPowerValue(RANGED_ATTACK) * 0.15f));
```
Comment (joschiwald) [inline src/server/game/Entities/Pet/Pet.cpp]: again, use values from dbc

File: src/server/game/Entities/Pet/Pet.cpp [hunk 4]
```diff
@@ -948,8 +949,20 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
             //damage range is then petlevel / 2
             SetBaseWeaponDamage(BASE_ATTACK, MAXDAMAGE, float(petlevel + (petlevel / 4)));
             //damage is increased afterwards as strength and pet scaling modify attack power
-            break;
+            if (GetAura(SPELL_WILD_HUNT_R2))
+            {
+                SetModifierValue(UNIT_MOD_STAT_STAMINA, BASE_VALUE, float(m_owner->GetStat(STAT_STAMINA)) * 0.4f);
+                SetBonusDamage(int32(GetOwner()->GetTotalAttackPowerValue(RANGED_ATTACK) * 0.3f));
+            }
+            else if (GetAura(SPELL_WILD_HUNT_R1))
```
Comment (joschiwald) [inline src/server/game/Entities/Pet/Pet.cpp]: these spells are not auras

**Comment by Keader** [comment]
Pet never has aura SPELL_WILD_HUNT_R2 and SPELL_WILD_HUNT_R1
It needs another way to check if he has the talent

File: src/server/game/Entities/Unit/StatSystem.cpp [hunk 5]
```diff
@@ -1146,18 +1146,9 @@ bool Guardian::UpdateStats(Stats stat)
...
-
-                if (itr != ToPet()->m_spells.end())                                 // If pet has Wild Hunt
-                {
-                    SpellInfo const* spellInfo = sSpellMgr->EnsureSpellInfo(itr->first); // Then get the SpellProto and add the dummy effect value
-                    AddPct(mod, spellInfo->Effects[EFFECT_0].CalcValue());
-                }
+                ownersBonus = float(owner->GetStat(stat)) * mod;
+                value += ownersBonus;
             }
-            ownersBonus = float(owner->GetStat(stat)) * mod;
```
Comment (Keader) [inline src/server/game/Entities/Unit/StatSystem.cpp]: This broken Gargoyle Pet (DK).

**Comment by Goatform** [comment]
@Keader, here you said  #6709 that this fix works for you right ?

File: src/server/game/Entities/Unit/StatSystem.cpp (hunk 5 above)
Comment (Goatform) [inline src/server/game/Entities/Unit/StatSystem.cpp]: @Keader. I didn't removed it.

All what I did in that part of code was:

``` diff
-            if (IsPet())
-            {
-                PetSpellMap::const_iterator itr = (ToPet()->m_spells.find(62758)); // Wild Hunt rank 1
-                if (itr == ToPet()->m_spells.end())
-                    itr = ToPet()->m_spells.find(62762);                            // Wild Hunt rank 2
-
-                if (itr != ToPet()->m_spells.end())                                 // If pet has Wild Hunt
-                {
-                    SpellInfo const* spellInfo = sSpellMgr->EnsureSpellInfo(itr->first); // Then get the SpellProto and add the dummy effect value
-                    AddPct(mod, spellInfo->Effects[EFFECT_0].CalcValue());
-                }
-            }
-            ownersBonus = float(owner->GetStat(stat)) * mod;
-            value += ownersBonus;
+           if (IsPet())
+           {
+               ownersBonus = float(owner->GetStat(stat)) * mod;
+               value += ownersBonus;
+           }

**Comment by sirikfoll** [comment]
This doesn't work.
By the time your code is called, the Pet don't have his talents loaded, so 
if `(GetAura(SPELL_WILD_HUNT_R2))` and `(GetAura(SPELL_WILD_HUNT_R1))` will never be true

**Comment by Aokromes** [comment]
/home/travis/build/TrinityCore/TrinityCore/src/server/game/Entities/Pet/Pet.cpp:924:13: fatal error: 
      enumeration value 'MAX_PET_TYPE' not handled in switch [-Wswitch]
    switch (petType)
            ^

**Comment by Keader** [comment]
@Goatform At the time, I believed it was working because another talent build survival had similar effects, giving a false impression of operation.
Basically what happens @sirikfoll  said.
And  Gargoyle is not a pet, then do not receive the bonus stamina.

**Comment by Goatform** [comment]
This will be fixed later.

**Pull request closed without merging**